AACT-439: As a user I should be able to edit a query that I created

Javier-Jimenez-18 opened the pull request:
- Added a button/link on the show page to /queries/:id/edit

<img width="1280" alt="Screen Shot 2022-12-05 at 3 36 13 PM" src="https://user-images.githubusercontent.com/81119399/205771808-ae280b28-4231-4761-99c8-2703de6624fc.png">

<img width="1280" alt="Screen Shot 2022-12-05 at 3 55 44 PM" src="https://user-images.githubusercontent.com/81119399/205771839-00d4c304-8f6a-4540-ad9a-176c7d1dfb24.png">

- Created an edit action in the Saved Queries controller that loads the query that we are editing. It checks to see if the currently logged in user created the query and if they did create it, then show the edit page.

<img width="1280" alt="Screen Shot 2022-12-05 at 3 40 26 PM" src="https://user-images.githubusercontent.com/81119399/205772874-e64b60f0-8bd9-4816-a22d-99a3600910b8.png">

<img width="1280" alt="Screen Shot 2022-12-05 at 3 40 42 PM" src="https://user-images.githubusercontent.com/81119399/205772915-7d46b4a3-b00a-4799-b3e0-4f2b68a6fa77.png">

- If the logged in user did not create the query, it redirects them to the index page for the saved queries and adds a flash message saying “You do not have permission to edit that query”. 

<img width="1280" alt="Screen Shot 2022-12-05 at 3 43 33 PM" src="https://user-images.githubusercontent.com/81119399/205773152-1fb552b5-90d3-4baf-b0ee-23e999d4111d.png">

<img width="1280" alt="Screen Shot 2022-12-05 at 3 44 18 PM" src="https://user-images.githubusercontent.com/81119399/205773185-5edc61e3-3572-4fca-b8b5-51d486d876a5.png">

- Updated SavedQueriesController RSpec Controller test file.

<img width="1280" alt="Screen Shot 2022-12-05 at 3 17 53 PM" src="https://user-images.githubusercontent.com/81119399/205773335-6a184f25-51dd-43de-9f64-e3b8d2d51280.png">

- The SavedQueriesController RSpec Request test file has some issues and will be investigated further.

Javier-Jimenez-18 committed:
Generated the QueriesController with action methods new and create.

Javier-Jimenez-18 committed:
Deleted migration CreateTableSavedQuery. Generated model SavedQuery and Spec files.

Javier-Jimenez-18 committed:
Added new and create action methods for the QueriesController. Created a form for the new action view page.

Javier-Jimenez-18 committed:
merged feat/AACT-435-create_controller_queries

Javier-Jimenez-18 committed:
Updated Queries _form and new view pages.

Javier-Jimenez-18 committed:
Updated _form view page for QueriesController. Updated to resources Queries routes.

Javier-Jimenez-18 committed:
Added attribute validations to savedQuery model.

Javier-Jimenez-18 committed:
Merge branch 'feat/AACT-434-create_model_SavedQuery' into feat/AACT-435-create_controller_queries

Javier-Jimenez-18 committed:
Modified QueriesController and view _form.

Javier-Jimenez-18 committed:
Renamed QueriesController to SavedQueriesController, views, route, and Spec file.

Javier-Jimenez-18 committed:
Added index action to SavedQueriesController and view page. Updated _form and new view pages.

Javier-Jimenez-18 committed:
Added the show action to the SavedQueriesController. Created the show view page.

Javier-Jimenez-18 committed:
Added show action to SavedQueriesController. Updated create action. Created show view page. Updated index view page to show username. Updated _form view page to add a checkbox to make SQL query public. Modified shared.scss stylesheet to remove label padding.

Javier-Jimenez-18 committed:
Added destroy action to SavedQueriesController. Updated the show view page. Updated application layout view.

Javier-Jimenez-18 committed:
Created edit and update actions for SavedQueriesController. Added display of SQL Query Public/Not Public and Edit button to show view page. Created edit view page.

Javier-Jimenez-18 committed:
Added has_many saved_queries to the User class. Modified and updated saved_queries view pages. Created saved_queries RSpec Requests test file.

Javier-Jimenez-18 committed:
Updated SavedQueriesController Rspec Controller and Request test files. The Request test file has some issues and will be investigated further.

Javier-Jimenez-18 committed:
Added more test case coverage to the SavedQueriesController RSpec Request test file.

kholtzman5610 reviewed (approved):
Good job with the test cases and screenshots. Looks good

micronix reviewed (changes requested):
Looks good, just some small changes

Javier-Jimenez-18 committed:
Modified SavedQueriesController index, edit, update, and destroy actions to check to see if the currently logged in user created the query. Updated RSpec Controller and Request test files. Cleaned up and refactored code.

Javier-Jimenez-18 commented:
- Modified SavedQueriesController index, edit, update, and destroy actions to check to see if the currently logged in user created the query. If not, it renders a not found error (404) page.
- Index shows all the queries that belong to the currently logged in user & all the queries that don't belong to the user and have Public set true. 

<img width="1280" alt="Screen Shot 2022-12-14 at 4 58 56 PM" src="https://user-images.githubusercontent.com/81119399/207753221-2897e1fa-5f44-4e1a-b3e8-fb224fd2437d.png">

<img width="1280" alt="Screen Shot 2022-12-14 at 4 59 57 PM" src="https://user-images.githubusercontent.com/81119399/207753250-abd07d0f-2b9e-4c34-8607-898f3ac8b5ce.png">

<img width="1280" alt="Screen Shot 2022-12-14 at 5 04 41 PM" src="https://user-images.githubusercontent.com/81119399/207753275-02d25b1a-c988-4dbf-9506-8b9506227f7a.png">

<img width="1280" alt="Screen Shot 2022-12-14 at 5 04 51 PM" src="https://user-images.githubusercontent.com/81119399/207753289-b1c406a5-115f-4b6c-baf5-85cba3530d89.png">

- Updated RSpec Controller test file. 

<img width="1280" alt="Screen Shot 2022-12-14 at 4 56 05 PM" src="https://user-images.githubusercontent.com/81119399/207753355-bd9a55ff-e277-41df-bb55-18bdcbb5e3b6.png">

- Updated RSpec Request test file. Each test case PASSES when run individually. Will have to investigate further to see why the test file has FAILURES when all test cases are run together.

- Cleaned up and refactored code.

Javier-Jimenez-18 commented:
All saved queries that are public, or that belong to the current user are shown in the index page. But, the current user can't go to the show page of another user's query, not_found (404) gets rendered. Is this what we want? I thought the current user can go to the show page of any public query and be able to at least run the SQL, but not be able to edit or delete the query. Is this correct?

Javier-Jimenez-18 committed:
Modified Show, Edit, Update, and Destroy actions for the SavedQueriesController. Allowed a logged-in user to view the show page of another user's public query. Modified and updated the RSpec Controller and Request test files.

Javier-Jimenez-18 commented:
- Modified Show, Edit, Update, and Destroy actions for the SavedQueriesController. Allowed a logged-in user to view the show page of another user's public query. 

<img width="1280" alt="Screen Shot 2022-12-16 at 1 51 36 AM" src="https://user-images.githubusercontent.com/81119399/208074970-6bf232f3-ffa8-44fc-8f4c-42320623123c.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 1 52 08 AM" src="https://user-images.githubusercontent.com/81119399/208075007-e5f8f5b6-da6d-4280-a7d6-427c6a66d4c8.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 1 53 31 AM" src="https://user-images.githubusercontent.com/81119399/208075026-bd5e7269-815b-45f4-a377-bc45fe2bdf23.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 1 55 12 AM" src="https://user-images.githubusercontent.com/81119399/208075069-ea731a2d-ed6c-4c5b-bca4-7b13da4bcc3a.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 1 55 25 AM" src="https://user-images.githubusercontent.com/81119399/208075093-0728d943-775b-464d-8cad-1a17d3271731.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 1 56 57 AM" src="https://user-images.githubusercontent.com/81119399/208075116-28e1004f-4223-4d00-ae25-837e0fe050cb.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 1 57 43 AM" src="https://user-images.githubusercontent.com/81119399/208075139-39da4ddd-ff29-4dea-a4e4-aa4e57ef31f5.png">

- Modified and updated the RSpec Controller test file.

<img width="1280" alt="Screen Shot 2022-12-16 at 1 50 22 AM" src="https://user-images.githubusercontent.com/81119399/208075234-bf70f15f-fd9a-4d0e-a3f4-fc7c801e0084.png">

- Modified and updated the RSpec Request test file. Each test case PASSES when run individually. Will have to investigate further to see why the test file has FAILURES when all test cases are run together.

micronix reviewed (changes requested):
The `show` action should have the logic to allow any user to see it if it is public or the user that created it is trying to see it.

The `edit`, `update` & `destroy` should only be accessible to the user that created it.

Javier-Jimenez-18 committed:
Modified Show, Edit, and Destroy actions in the SavedQueriesController. Modified Show view page to only show edit and delete buttons for the User that created the query.

Javier-Jimenez-18 commented:
- Modified Show, Edit, and Destroy actions in the SavedQueriesController. 

- Modified Show view page to only show edit and delete buttons for the User that created the query.

<img width="1280" alt="Screen Shot 2022-12-16 at 10 12 29 AM" src="https://user-images.githubusercontent.com/81119399/208163182-9e808359-7317-46cc-9749-a414f07446d3.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 10 16 52 AM" src="https://user-images.githubusercontent.com/81119399/208163199-d3e4c0c5-f675-4b2d-b3a0-bf44150ba566.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 10 13 01 AM" src="https://user-images.githubusercontent.com/81119399/208163217-065692ee-32e1-4dcc-89f8-033d70a95d2d.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 10 13 37 AM" src="https://user-images.githubusercontent.com/81119399/208163251-d6c61e16-a635-4e87-8d53-3427ce00edc8.png">

<img width="1280" alt="Screen Shot 2022-12-16 at 9 15 38 AM" src="https://user-images.githubusercontent.com/81119399/208163285-e99250e8-6226-4347-9cfb-43050cf6ba71.png">
